### ISSUE TITLE
TypeError happened in `retrieve_job` function

### ISSUE THREAD

**Issue by snuffkin** [Qiskit/qiskit-ibmq-provider#223]
### Information

- **Qiskit IBMQ Provider version**:0.2.2
- **API URL or version**:https://api.quantum-computing.ibm.com/api/Hubs/ibm-q/Groups/open/Projects/main

- **Python version**:3.7
- **Operating system**:Ubuntu18.04

### What is the current behavior?
When I called `retrieve_job` function, the following error happens:
```
File "/opt/python/qiskit/providers/ibmq/api/ibmqconnector.py", line 196, in get_job
   if ('result' in qasm) and ('data' in qasm['result']):
TypeError: argument of type 'NoneType' is not iterable
```

### Steps to reproduce the problem
If `qasm['result']` is None, this problem happens.

### What is the expected behavior?
the error doesn't happen.

### Suggested solutions
current code:
`if ('result' in qasm) and ('data' in qasm['result']):`

I think it should be this:
`if ('result' in qasm) and (qasm['result'] is not None) and ('data' in qasm['result']):`

**Comment by jyu00** [comment]
Hi @snuffkin, thank you reporting this bug! It's been fixed by #154 and will be available in the next release.

**Comment by snuffkin** [comment]
Hi @jyu00 , thank you!
